test/e2e/internal/notation/host.go

```diff
@@ -0,0 +1,109 @@
+package notation
+
+import (
+	"os"
+	"path/filepath"
+
+	"github.com/notaryproject/notation/test/e2e/internal/utils"
+)
+
+// Host creates a virtualized notation testing host by modify
+// the "XDG_CONFIG_HOME" environment variable of the Executor.
+//
+// options is the required testing environment options
+// fn is the callback function containing the testing logic.
+func Host(options []utils.HostOption, fn func(notation *utils.ExecOpts, artifact *Artifact, vhost *utils.VirtualHost)) {
+	// create a vhost
+	vhost, err := utils.NewVirtualHost(NotationBinPath, CreateNotationDirOption())
+	if err != nil {
+		panic(err)
+	}
+
+	// set additional options
+	vhost.SetOption(options...)
+
+	// generate a repository with an artifact
+	artifact := GenerateArtifact()
+
+	// run the main logic
+	fn(vhost.Executor, artifact, vhost)
+
+	// remove the generated repository and artifact
+	if err := artifact.Remove(); err != nil {
+		panic(err)
+	}
+}
+
+// Opts is a grammar sugar to generate a list of HostOption
```

Comment:
```suggestion
// Opts is a grammar sugar to generate a list of HostOption.
```
Nit: most of the comments end up with a `.`

Reply:
Updated.